tsdb/ooo_head_read_test.go

```diff
@@ -879,26 +881,38 @@ func TestOOOHeadChunkReader_Chunk(t *testing.T) {
 //   - Response B comes from : Series(), in parallel new samples added to the head, then Chunk()
 //   - A == B
 func TestOOOHeadChunkReader_Chunk_ConsistentQueryResponseDespiteOfHeadExpanding(t *testing.T) {
+	for name, scenario := range sampleTypeScenarios {
+		t.Run(name, func(t *testing.T) {
+			testOOOHeadChunkReader_Chunk_ConsistentQueryResponseDespiteOfHeadExpanding(t, scenario)
+		})
+	}
+}
+
+// TestOOOHeadChunkReader_Chunk_ConsistentQueryResponseDespiteOfHeadExpanding tests
+// that if a query comes and performs a Series() call followed by a Chunks() call
+// the response is consistent with the data seen by Series() even if the OOO
+// head receives more samples before Chunks() is called.
+// An example:
+//   - Response A comes from: Series() then Chunk()
+//   - Response B comes from : Series(), in parallel new samples added to the head, then Chunk()
+//   - A == B
+//
```

Comment:
I think we can remove this comment because it's already commented in TestOOOHeadChunkReader_Chunk_ConsistentQueryResponseDespiteOfHeadExpanding